bsb/connectivity/point_cloud/geometric_shapes.py

```diff
@@ -0,0 +1,1205 @@
+from __future__ import annotations
+import numpy as np
+from scipy.spatial.transform import Rotation as R
+import pickle
+import abc
+import copy
+from typing import List, Tuple
+import numpy
+
+from .cloud_mesh_utils import (
+    rotate_3d_mesh_by_vec,
+    translate_3d_mesh_by_vec,
+    rotate_3d_mesh_by_rot_mat,
+)
+
+
+class GeometricShape(abc.ABC):
```

Comment:
```suggestion
@config.dynamic(attr_name="shape", auto_classmap=True, classmap_entry=None)
class GeometricShape(abc.ABC):
```

Parent needs to be a dynamic node, set classmap entry to None so that the base class is "abstract" and not directly invokable